rotkehlchen/exchanges/manager.py

```diff
@@ -82,58 +88,88 @@ def edit_exchange(
-                api_secret=api_secret,
-                passphrase=passphrase,
-                kraken_account_type=kraken_account_type,
-                binance_selected_trade_pairs=binance_selected_trade_pairs,
-                ftx_subaccount=ftx_subaccount,
-            )
-            if success is False:
-                # by raising we also rollback the DB writes due to user_write()
-                raise InputError(msg)
+        try:
+            with self.database.user_write() as cursor:
+                self.database.edit_exchange(
+                    cursor,
+                    name=name,
+                    location=location,
+                    new_name=new_name,
+                    api_key=api_key,
+                    api_secret=api_secret,
+                    passphrase=passphrase,
+                    kraken_account_type=kraken_account_type,
+                    binance_selected_trade_pairs=binance_selected_trade_pairs,
+                    ftx_subaccount=ftx_subaccount,
+                )
+        except InputError as e:
+            exchangeobj.reset_to_db_credentials()  # DB is already rolled back at this point
+            exchangeobj.reset_to_db_extras()
+            return False, f'Coudnt update exchange properties in the DB. {str(e)}'
 
-            return True, ''
+        # Finally edit the name of the exchange object
+        if new_name is not None:
+            exchangeobj.edit_exchange_name(new_name)
+
+        return True, ''
+
+    def delete_exchange(self, name: str, location: Location) -> tuple[bool, str]:
+        """
+        Deletes an exchange with the specified name + location from both connected_exchanges
+        and the DB.
+        """
+        if self.get_exchange(name=name, location=location) is None:
+            return False, f'{str(location)} exchange {name} is not registered'
 
-    def delete_exchange(self, name: str, location: Location) -> None:
-        """Deletes exchange only from the manager. Not from the DB"""
         exchanges_list = self.connected_exchanges.get(location)
         if exchanges_list is None:
-            return
+            return False, f'{str(location)} exchange {name} is not registered'
 
-        if len(exchanges_list) == 1:
+        last_in_location = len(exchanges_list) == 1
+        if last_in_location is True:
             self.connected_exchanges.pop(location)
-            return
-
-        self.connected_exchanges[location] = [x for x in exchanges_list if x.name != name]
+        else:
+            self.connected_exchanges[location] = [x for x in exchanges_list if x.name != name]
+        with self.database.user_write() as write_cursor:  # Also remove it from the db
+            self.database.remove_exchange(write_cursor=write_cursor, name=name, location=location)  # noqa: E501
+            if last_in_location is True:
+                self.database.delete_used_query_range_for_exchange(write_cursor=write_cursor, location=location)  # noqa: E501
```

Comment:
I'm not sure why we do it like this. When deleting an exchange seems that we don't remove the used query ranges for that exchange until all the exchanges for that location have been removed. Maybe we can improve this in another PR? Take note of this please